gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/resources/IosResources.kt

```diff
@@ -81,19 +81,24 @@ internal fun Project.configureSyncIosComposeResources(
     plugins.withId(COCOAPODS_PLUGIN_ID) {
         (kotlinExtension as ExtensionAware).extensions.getByType(CocoapodsExtension::class.java).apply {
             framework { podFramework ->
-                val syncDir = podFramework.getFinalResourcesDir().get().asFile.relativeTo(projectDir)
-                val specAttr = "['${syncDir.path}']"
-                extraSpecAttributes["resources"] = specAttr
-                project.tasks.named("podInstall").configure {
-                    it.doFirst {
-                        if (extraSpecAttributes["resources"] != specAttr) {
-                            error("""
+                // DIRTY HACK: we don't have an API to configure PodspecTask.
+                // We need to use the cocoapods DSL only but to provide a correct resources path we need lazy read a framework baseName.
+                // Which may be configured be the same DSL
+                podFramework.project.afterEvaluate {
```

Comment:
I'd extract this apply/framework/afterEvaluate chain to separate function to localize this "hack" there and reduce nesting blocks here.
Just suggestion/nitpick, not blocking